mmv1/products/netapp/activeDirectory.yaml

```
@@ -0,0 +1,252 @@
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      LOCATION_DESCRIPTION
+  - !ruby/object:Api::Type::String
+    name: 'active_directory_id'
+    description:
+      'ID of the active directory to create
+      `[_a-zA-Z0-9][-_.a-zA-Z0-9]*`.'
+    required: true
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'Active_Directory_Create'
+    primary_resource_id: 'test_active_directory'
+    vars:
+      active_directory_name: 'test-active-directory'
+properties:
+  # Fields go here
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      The resource name of the active directory. Format: projects/{projectNumber}/locations/{locationId}/activeDirectories/{activeDirectoryId}.
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    description: |
+      Create time of the active directory. A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    description: |
+      The state of the AD.
+    values:
+      - :STATE_UNSPECIFIED
+      - :CREATING
+      - :READY
+      - :UPDATING
+      - :DELETING
+      - :IN_USE
+      - :ERROR
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'domain'
+    description: |
+      Name of the Active Directory domain
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'site'
+    description: |
+      The Active Directory site the service will limit Domain Controller discovery too.
+    required: false
+  - !ruby/object:Api::Type::String
+    name: 'dns'
+    description: |
+      Comma separated list of DNS server IP addresses for the Active Directory domain.
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'netBiosPrefix'
+    description: |
+      NetBIOSPrefix is used as a prefix for SMB server name.
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'organizationalUnit'
+    description: |
+      The Organizational Unit (OU) within the Windows Active Directory the user belongs to.
+    required: false
+    default_from_api: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'aesEncryption'
+    description: |
+      If enabled, AES encryption will be enabled for SMB communication.
+    required: false
+  - !ruby/object:Api::Type::String
+    name: 'username'
+    description: |
+      Username of the Active Directory domain administrator.
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'password'
+    description: |
+      Password of the Active Directory domain administrator.
+    required: true
+    sensitive: true
```

```suggestion
    sensitive: true
    ignore_read: true
```
it looks like there is an issue due to the API request returning an asterisk filled value -- we can try including this ignore_read tag for now

---

Let me elaborate on this:
password is sensitive. So the API doesn't return the password. It returns "******************" instead, which it 18 times the "*" character.
Now comes the trick: If you send a password which is "******************", it basically means "please keep the old password which is stored and ignore the password I sent.
ignore_read is a double-edged sword, right? On one side, it will ignore the password the API returns to avoid fake configuration drift. On the other side, what if the user changes the password really and wants to update the AD resource? Wouldn't a TF apply silently ignore the changed password?